### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by brendt**

### CODE REVIEW

File: src/Projectionist.php [hunk 1]
```diff
@@ -53,6 +53,13 @@ public function addProjector($projector): Projectionist
         return $this;
     }
 
+    public function allEventHandlers(): EventHandlerCollection
+    {
+        $eventHandlers = $this->projectors->all()->merge($this->reactors->all())->toArray();
```
Comment (brendt) [inline src/Projectionist.php:58]: Why first calling `toArray` and then wrap it in a new `EventHandlerCollection` ? 

```
$this->projectors->all()->merge($this->reactors->all())
```

This already is an `EventHandlerCollection`

File: src/EventHandlers/EventHandlerCollection.php [hunk 2]
```diff
@@ -45,6 +49,35 @@ public function call(string $method)
     public function remove(array $eventHandlerClassNames): void
     {
         $this->eventHandlers = $this->eventHandlers
-            ->reject(fn (EventHandler $eventHandler) => in_array(get_class($eventHandler), $eventHandlerClassNames));
+            ->reject(
+                fn (EventHandler $eventHandler) => in_array(get_class($eventHandler), $eventHandlerClassNames)
+            );
+    }
+
+    public function sycnEventHandlers(): self
```
Comment (brendt) [inline src/EventHandlers/EventHandlerCollection.php:57]: Love these collection methods!

File: src/Projectionist.php (hunk 1 above)
Comment (freekmurze) [reply src/Projectionist.php:58]: Currently `EventHandlerCollection` does not extend from `Collection` and has no `merge` function.

Hold my Gini while I refactor some more.

File: src/EventHandlers/EventHandlerCollection.php (hunk 2 above)
Comment (freekmurze) [reply src/EventHandlers/EventHandlerCollection.php:57]: Me too!

**Pull request merged**